### PR TITLE
Exclude operators

### PR DESCRIPTION
**Pull request by Alcolo47**
Adding exclude-operators options because there is irrelevant operators ( < -> !=  is -> ==)

### CODE REVIEW

**Comment by abingham** [comment]
Thanks for your work on this. This kind of feature is certainly a useful one, and in fact we've had [some discussion along these lines already](https://github.com/sixty-north/cosmic-ray/issues/83). Also see the the skipping we implement in `comparison_operator_replacement.py`.

However, I think we need to take a different approach to implementing it. I would like to see this implemented as an *interceptor* rather than baked in to the core operator handling code. I know the interceptor feature isn't very well documented, and I apologize for that - my time for CR is very limited right now. But I do think it's the right way to filter/skip operators.

Here's how I see it working:
1. Create an interceptor plugin that is responsible for finding mutations that should be skipped (i.e. using regular expressions as you've done or via some sort of configuration)
2. When it runs, it can scan the work-db for jobs that should not run and mark them as "skipped".
3. Its configuration will be via the normal configuration file (which it can access from the work-db during interceptor execution time).
4. Users can configure the interceptor and en/disable it using the standard configuration system.

You can look at `cosmic_ray.interceptors.spor` for an example of how this could work. 

I think you'll be able to re-use the core of what you've already written, but the interceptor-based implementation won't require touching nearly as many files. It will be nicely encapsulated inside the existing system.

If I have some time this weekend I'll see if I can sketch this out more completely for you. Let me know if you have any questions. And again, thanks for your work on this...hopefully I'll have some time soon to give CR more attention!

**Comment by abingham** [comment]
> If I have some time this weekend I'll see if I can sketch this out more completely for you.

Based on your other PRs, I can see you're way ahead of me on understanding interceptors!

File: src/cosmic_ray/commands/init.py [hunk 1]
```diff
@@ -56,7 +57,27 @@ def init(module_paths, work_db, config):
       work_db: A `WorkDB` instance into which the work orders will be saved.
       config: The configuration for the new session.
     """
-    operator_names = cosmic_ray.plugins.operator_names()
+
+    operator_names = list(cosmic_ray.plugins.operator_names())
+
+    if log.isEnabledFor(logging.INFO):
```
Comment (abingham) [inline src/cosmic_ray/commands/init.py]: Why do we need this here? Isn't this what the "operators" command is for?

Comment (abingham) [reply src/cosmic_ray/commands/init.py]: With the exception of this logging, I think this PR looks good.

Comment (Alcolo47) [reply src/cosmic_ray/commands/init.py]: I don't understand the problem:
- Just this test? It is useless to build complex log trace is there is no log watchers.
- The entire log block: It's just cosmetic, but to see and understand all available operators, some classification seems useful.

Comment (abingham) [reply src/cosmic_ray/commands/init.py]: My point is that we've already got the `cosmic-ray operators` command which does this. If you want to update the output format of `operators`, I'm open to that discussion. But this logging just seems to duplicate an existing feature unnecessarily.

Comment (Alcolo47) [reply src/cosmic_ray/commands/init.py]: Ok , I have forgotten `cosmic-ray operators`. I will rollback this log

**Comment by abingham** [comment]
This is great, thanks!

**Comment by abingham** [comment]
I've merge this in as is, but I'd really like it if you could add a few tests. We just need some way to know if this functionality works and to detect regressions.

**Pull request merged**